src/main/java/org/myrobotlab/service/abstracts/AbstractServo.java

```diff
@@ -0,0 +1,764 @@
+
+    if (lastPos == newPos) {
+      lastPos = targetPos;
+      broadcastState(); // publishServo vs publishServoData .. GAH ! :P
+    }
+
+  }
+
+  @Override
+  public void moveTo(Float newPos) {
+    moveTo(newPos.doubleValue());
+  }
+
+  @Override
+  public void moveTo(Integer newPos) {
+    moveTo(newPos.doubleValue());
+  }
+
+  @Override
+  public void moveToBlocking(Double pos) {
+    moveTo(pos, true, null);
+  }
+
+  @Override
+  public void moveToBlocking(Double newPos, Long timeoutMs) {
+    moveTo(newPos, true, timeoutMs);
+  }
+
+  @Override
+  public void moveToBlocking(Float newPos) {
+    moveTo(newPos.doubleValue(), true, null);
+  }
+
+  @Override
+  public void moveToBlocking(Float newPos, Long timeoutMs) {
+    moveTo(newPos.doubleValue(), true, timeoutMs);
+  }
+
+  @Override
+  public void moveToBlocking(Integer newPos) {
+    moveTo(newPos.doubleValue(), true, null);
+  }
+
+  @Override
+  public void moveToBlocking(Integer newPos, Long timeoutMs) {
+    moveTo(newPos.doubleValue(), true, timeoutMs);
+  }
+
+  @Override
+  public ServoControl publishMoveTo(ServoControl sc) {
+    return sc;
+  }
+
+  @Override
+  public ServoData publishServoData(ServoStatus status, Double pos) {
+    ServoData sd = new ServoData();
+    sd.name = getName();
+    sd.pos = pos;
+    sd.speed = speed;
+    sd.targetPos = targetPos;
+    sd.state = status;
+    return sd;
+  }
+
+  /**
+   * will disable then detach this servo from all controllers
+   */
+  public void releaseService() {
+    super.releaseService();
+    detach();
+  }
+
+  @Override
+  public void rest() {
+    targetPos = rest;
+    for (String controller : controllers) {
+      ServiceInterface si = Runtime.getService(controller);
+      if (si.isLocal()) {
+        ((ServoController) Runtime.getService(controller)).servoMoveTo(this);
+      } else {
+        send(controller, "moveTo", this);
+      }
+    }
+  }
+
+  @Override
+  public void setAcceleration(Double acceleration) {
+    this.acceleration = acceleration;
+  }
+
+  @Override
+  public void setAcceleration(Integer acceleration) {
+    this.acceleration = acceleration.doubleValue();
+  }
+
+  @Override
+  public void setAutoDisable(Boolean autoDisable) {
+    this.autoDisable = autoDisable;
+  }
+
+  @Override
+  public void setInverted(Boolean invert) {
+    mapper.setInverted(invert);
+  }
+
+  @Override
+  public void setMinMax(Double min, Double max) {
+    mapper.setMinMaxInput(min, max);
+  }
+
+  @Override
+  public void setMinMax(Integer min, Integer max) {
+    mapper.setMinMaxInput(min, max);
+  }
+
+  @Override
+  public void setPin(Integer pin) {
+    this.pin = pin + "";
+  }
+
+  @Override
+  public void setPin(String pin) {
+    this.pin = pin;
+  }
+
+  @Override
+  public void setRest(Double rest) {
+    this.rest = rest;
+  }
+
+  @Override
+  public void setRest(Integer rest) {
+    this.rest = rest.doubleValue();
+  }
+
+  @Override
+  public void setSpeed(Double degreesPerSecond) {
```

Comment:
I think we need to maintain the backwards compatibility for the old skewl setSpeed . our new proper method is setVelocity which takes a double that specifies  the angular velocity of the servo that we will try to achieve.